python/cudf_polars/cudf_polars/experimental/parallel.py

```diff
@@ -234,3 +255,61 @@ def evaluate_dask(ir: IR) -> DataFrame:
 
     graph, key = task_graph(ir, partition_info)
     return get(graph, key)
+
+
+def _concat(dfs: Sequence[DataFrame]) -> DataFrame:
+    # Concatenate a sequence of DataFrames vertically
+    return Union.do_evaluate(None, *dfs)
+
+
+##
+## Union
+##
+
+
+@lower_ir_node.register(Union)
+def _(
+    ir: Union, rec: LowerIRTransformer
+) -> tuple[IR, MutableMapping[IR, PartitionInfo]]:
+    # zlice must be None
+    if ir.zlice is not None:
+        return rec.state["default_mapper"](ir)  # pragma: no cover
+
+    # Lower children
+    children, _partition_info = zip(*(rec(c) for c in ir.children), strict=False)
```

Comment:
```suggestion
    children, _partition_info = zip(*(rec(c) for c in ir.children), strict=True)
```
All the pieces should be the same length, I think.